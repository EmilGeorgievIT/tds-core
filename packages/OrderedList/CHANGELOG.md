# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [2.0.17](https://github.com/telusdigital/tds/compare/@tds/core-ordered-list@2.0.16...@tds/core-ordered-list@2.0.17) (2019-04-22)

**Note:** Version bump only for package @tds/core-ordered-list





## [2.0.16](https://github.com/telusdigital/tds/compare/@tds/core-ordered-list@2.0.15...@tds/core-ordered-list@2.0.16) (2019-04-11)


### Bug Fixes

* **core-ordered-list:** use className prop for Box ([d998464](https://github.com/telusdigital/tds/commit/d998464))





## [2.0.15](https://github.com/telusdigital/tds/compare/@tds/core-ordered-list@2.0.14...@tds/core-ordered-list@2.0.15) (2019-04-08)


### Bug Fixes

* **core-ordered-list:** allowed conditional children rendering ([ab2ec94](https://github.com/telusdigital/tds/commit/ab2ec94))
* **core-ordered-list:** converted children to array before filtering ([2bca8a0](https://github.com/telusdigital/tds/commit/2bca8a0))





## [2.0.14](https://github.com/telusdigital/tds/compare/@tds/core-ordered-list@2.0.13...@tds/core-ordered-list@2.0.14) (2019-04-04)

**Note:** Version bump only for package @tds/core-ordered-list





## [2.0.13](https://github.com/telusdigital/tds/compare/@tds/core-ordered-list@2.0.12...@tds/core-ordered-list@2.0.13) (2019-03-28)

**Note:** Version bump only for package @tds/core-ordered-list





## [2.0.12](https://github.com/telusdigital/tds/compare/@tds/core-ordered-list@2.0.11...@tds/core-ordered-list@2.0.12) (2019-03-28)

**Note:** Version bump only for package @tds/core-ordered-list





## [2.0.11](https://github.com/telusdigital/tds/compare/@tds/core-ordered-list@2.0.10...@tds/core-ordered-list@2.0.11) (2019-03-20)

**Note:** Version bump only for package @tds/core-ordered-list





## [2.0.10](https://github.com/telusdigital/tds/compare/@tds/core-ordered-list@2.0.9...@tds/core-ordered-list@2.0.10) (2019-03-20)

**Note:** Version bump only for package @tds/core-ordered-list





## [2.0.9](https://github.com/telusdigital/tds/compare/@tds/core-ordered-list@2.0.8...@tds/core-ordered-list@2.0.9) (2019-03-18)

**Note:** Version bump only for package @tds/core-ordered-list





## [2.0.8](https://github.com/telusdigital/tds/compare/@tds/core-ordered-list@2.0.7...@tds/core-ordered-list@2.0.8) (2019-03-18)

**Note:** Version bump only for package @tds/core-ordered-list





## [2.0.7](https://github.com/telusdigital/tds/compare/@tds/core-ordered-list@2.0.6...@tds/core-ordered-list@2.0.7) (2019-03-11)

**Note:** Version bump only for package @tds/core-ordered-list





## [2.0.6](https://github.com/telusdigital/tds/compare/@tds/core-ordered-list@2.0.5...@tds/core-ordered-list@2.0.6) (2019-03-05)

**Note:** Version bump only for package @tds/core-ordered-list





## [2.0.5](https://github.com/telusdigital/tds/compare/@tds/core-ordered-list@2.0.4...@tds/core-ordered-list@2.0.5) (2019-02-06)

**Note:** Version bump only for package @tds/core-ordered-list





## [2.0.4](https://github.com/telusdigital/tds/compare/@tds/core-ordered-list@2.0.3...@tds/core-ordered-list@2.0.4) (2019-01-10)

**Note:** Version bump only for package @tds/core-ordered-list

<a name="2.0.3"></a>

## [2.0.3](https://github.com/telusdigital/tds/compare/@tds/core-ordered-list@2.0.2...@tds/core-ordered-list@2.0.3) (2018-12-19)

**Note:** Version bump only for package @tds/core-ordered-list

<a name="2.0.2"></a>

## [2.0.2](https://github.com/telusdigital/tds/compare/@tds/core-ordered-list@2.0.1...@tds/core-ordered-list@2.0.2) (2018-12-10)

**Note:** Version bump only for package @tds/core-ordered-list

<a name="2.0.1"></a>

## [2.0.1](https://github.com/telusdigital/tds/compare/@tds/core-ordered-list@2.0.0...@tds/core-ordered-list@2.0.1) (2018-12-06)

**Note:** Version bump only for package @tds/core-ordered-list

<a name="2.0.0"></a>

# [2.0.0](https://github.com/telusdigital/tds/compare/@tds/core-ordered-list@1.0.1...@tds/core-ordered-list@2.0.0) (2018-11-01)

### Bug Fixes

- **core-ordered-list:** adjust nested list spacing ([a9fccbc](https://github.com/telusdigital/tds/commit/a9fccbc))
- **core-ordered-list:** replace airbnb-prop-types with util-prop-types ([777c8cf](https://github.com/telusdigital/tds/commit/777c8cf))

### Features

- **core-ordered-list:** add [@tds](https://github.com/tds)/core-text as dependency ([72882d2](https://github.com/telusdigital/tds/commit/72882d2))
- **core-ordered-list:** handle nested list spacing, ordered list size ([8bfd3e5](https://github.com/telusdigital/tds/commit/8bfd3e5))
- **core-ordered-list:** update styles, add sizing prop ([9b80e6f](https://github.com/telusdigital/tds/commit/9b80e6f))

### BREAKING CHANGES

- **core-ordered-list:** add size prop to ordered list

<a name="1.0.1"></a>

## [1.0.1](https://github.com/telusdigital/tds/compare/@tds/core-ordered-list@1.0.0...@tds/core-ordered-list@1.0.1) (2018-04-02)

### Bug Fixes

- **core-box:** update tests ([c0f0343](https://github.com/telusdigital/tds/commit/c0f0343))
