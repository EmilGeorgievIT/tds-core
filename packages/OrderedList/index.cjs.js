require('./dist/index.css')
const OrderedList = require('./dist/index.cjs')

module.exports = OrderedList
