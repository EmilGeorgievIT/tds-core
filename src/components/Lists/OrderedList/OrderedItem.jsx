import React from 'react'
import PropTypes from 'prop-types'

import safeRest from '../../../safeRest'

import textStyles from '../../Typography/Text/Text.modules.scss'
import styles from '../ListItem.modules.scss'

const OrderedItem = ({ size, children, ...rest }) => {
  const classes = `
    ${styles.item}
    ${textStyles[size]}
  `

  return (
    <li {...safeRest(rest)} className={classes}>
      {children}
    </li>
  )
}

OrderedItem.propTypes = {
  size: PropTypes.oneOf([
    'medium',
    'large'
  ]),
  children: PropTypes.node.isRequired
}

OrderedItem.defaultProps = {
  size: undefined
}

OrderedItem.displayName = 'OrderedList.Item'

export default OrderedItem
