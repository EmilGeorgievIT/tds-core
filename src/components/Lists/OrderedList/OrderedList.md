```
<OrderedList size="large">
  <OrderedList.Item>Nunc condimentum non est sed rutrum. Donec porta torto mattis velit fermentum iaculis</OrderedList.Item>
  <OrderedList.Item>Lorem ipsum dolor sit amet, consectetur adipiscing elit
    <OrderedList variant="capital">
      <OrderedList.Item>Quisque tincidunt dolor</OrderedList.Item>
      <OrderedList.Item>Nunc condimentum non est sed rutrum. Donec porta tortor mattis velit</OrderedList.Item>
    </OrderedList>
  </OrderedList.Item>
  <OrderedList.Item>Optik High Speed Turbo</OrderedList.Item>
  <OrderedList.Item>High Speed Turbo 25</OrderedList.Item>
</OrderedList>
```
