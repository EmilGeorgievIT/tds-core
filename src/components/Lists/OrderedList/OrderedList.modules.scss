.base {
  padding-left: 40px;
}

.decimal {
  composes: base;

  list-style-type: decimal;
}

.upperAlpha {
  composes: base;

  list-style-type: upper-alpha;
}

.lowerAlpha {
  composes: base;

  list-style-type: lower-alpha;
}
