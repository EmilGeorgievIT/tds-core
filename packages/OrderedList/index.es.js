import './dist/index.css'
import OrderedList from './dist/index.es'

export default OrderedList
