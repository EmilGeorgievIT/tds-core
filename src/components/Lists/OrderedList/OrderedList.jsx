import React from 'react'
import PropTypes from 'prop-types'
import { childrenOfType } from 'airbnb-prop-types'

import safeRest from '../../../utils/safeRest'
import OrderedItem from './OrderedItem'

import styles from './OrderedList.modules.scss'

/**
 * <span class="docs--badge__wip">wip</span>
 */
const OrderedList = ({ listStyle, children, ...rest }) => (
  <ol {...safeRest(rest)} className={styles[listStyle]}>
    {children}
  </ol>
)

OrderedList.propTypes = {
  /**
   * The bullet style.
   */
  listStyle: PropTypes.oneOf([
    'decimal',
    'upperAlpha',
    'lowerAlpha'
  ]),
  /**
   * The list items. Must be at least one `OrderedList.Item`.
   */
  children: childrenOfType(OrderedItem).isRequired
}

OrderedList.defaultProps = {
  listStyle: 'decimal'
}

OrderedList.Item = OrderedItem

export default OrderedList
