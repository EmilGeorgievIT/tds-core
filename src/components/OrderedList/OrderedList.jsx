import React from 'react'
import PropTypes from 'prop-types'

import safeRest from '../../safeRest'
import OrderedItem from './OrderedItem/OrderedItem'

import styles from './OrderedList.modules.scss'

const OrderedList = ({ variant, size, children, ...rest }) => {
  const classes = `
    ${styles.base}
    ${styles[variant]}
    ${styles[size]}
  `
  return (
    <ol {...safeRest(rest)} className={classes}>
      {children}
    </ol>
  )
}

OrderedList.propTypes = {
  /**
   * The type of list.
   */
  variant: PropTypes.oneOf([
    'number',
    'capital',
    'lowercase'
  ]),
  size: PropTypes.oneOf([
    'medium',
    'large'
  ]),
  children: PropTypes.node.isRequired
}

OrderedList.defaultProps = {
  variant: 'number',
  size: 'medium'
}

OrderedList.Item = OrderedItem

export default OrderedList
