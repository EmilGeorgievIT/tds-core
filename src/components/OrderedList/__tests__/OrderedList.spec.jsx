import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import OrderedList from '../OrderedList'

describe('<OrderedList />', () => {
  const defaultChildren = '<li>One</li><li>Two</li>'
  const doShallow = (props = {}, children = defaultChildren) => (
    shallow(<OrderedList {...props}>{children}</OrderedList>)
  )

  it('renders', () => {
    const orderedList = doShallow()

    expect(toJson(orderedList)).toMatchSnapshot()
  })

  it('renders an HTML ul tag', () => {
    const orderedList = doShallow()

    expect(orderedList).toHaveTagName('ol')
  })

  it('can have a variant', () => {
    let orderedList = doShallow({ variant: undefined })
    expect(orderedList).toHaveClassName('number')

    orderedList = doShallow({ variant: 'capital' })
    expect(orderedList).toHaveClassName('capital')
  })


  it('can be sized', () => {
    let orderedList = doShallow()
    expect(orderedList).toHaveClassName('medium')

    orderedList = doShallow({ size: 'large' })
    expect(orderedList).toHaveClassName('large')
  })
})
