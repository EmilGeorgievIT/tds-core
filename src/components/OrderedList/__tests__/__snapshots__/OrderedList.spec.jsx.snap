// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OrderedList /> renders 1`] = `
<ol
  className="
    base
    number
    medium
  "
>
  &lt;li&gt;One&lt;/li&gt;&lt;li&gt;Two&lt;/li&gt;
</ol>
`;
